src/monster.h

```diff
@@ -419,8 +419,7 @@ class monster : public Creature
         tripoint wander_pos; // Wander destination - Just try to move in that direction
         int wandf;           // Urge to wander - Increased by sound, decrements each move
         std::vector<item> inv; // Inventory
-        player *dragged_foe; // player being dragged by the monster
-        int dragged_foe_id; // id of player being dragged by the monster (for save/load)
+        int dragged_foe_id = -1; // id of player being dragged by the monster
```

Comment:
While you're at it, could you please also fix the inconsistency as to what number is used for invalid id? For example, the following comment says that the invalid id should be 0:
https://github.com/CleverRaven/Cataclysm-DDA/blob/cdbd895fd439ee51ecb55048a416f0be14a0e201/src/player.h#L1645-L1646
The code in `game.cpp` also suggests so:
https://github.com/CleverRaven/Cataclysm-DDA/blob/cdbd895fd439ee51ecb55048a416f0be14a0e201/src/game.cpp#L3736-L3741
`next_npc_id` is initialized as 1 in a new game:
https://github.com/CleverRaven/Cataclysm-DDA/blob/cdbd895fd439ee51ecb55048a416f0be14a0e201/src/game.cpp#L270
And `assign_npc_id()` is used to assign id to newly created npcs and the player.
However, the following code uses -1 as the invalid number instead:
https://github.com/CleverRaven/Cataclysm-DDA/blob/cdbd895fd439ee51ecb55048a416f0be14a0e201/src/player.cpp#L10698-L10707
I think fixing `player::setID()` should be enough to address this problem

Reply:
I would love for this inconsistency to be fixed, but I'm concerned about it creeping the scope of this PR, especially since this addresses a bug that's causing segfaults on game save. The last thing I want is to add extra complexity, and addressing IDs for player objects could be done as a separate PR.

As it stands I believe all code will treat `-1` as an invalid ID, even if there's some debate as to whether `0` is valid or not.

Reply:
Fair enough. It's not like the inconsistency has caused any problem since it has been there for at least 5 years...

Reply:
> could you please also fix the inconsistency as to what number is used for invalid id? 

And that is why you don't use sentinels. Instead use `std::optional<int>`, which makes it perfectly clear to the reader when it's valid (when the optional contains an actual `int`) and when not (when the optional is empty).

Reply:
Marking this as resolved. I'm sure a separate pull-request add proper optional types rather than using sentinels would be appreciated.